lib/LANraragi/Plugin/Metadata/MEMS.pm

```diff
@@ -0,0 +1,161 @@
+package LANraragi::Plugin::Metadata::MEMS;
+
+use strict;
+use warnings;
+use Mojo::UserAgent;
+use Mojo::JSON qw(decode_json encode_json);
+use LANraragi::Utils::Logging qw(get_logger);
+
+# Meta-information about the plugin.
+sub plugin_info {
+  return (
+    # Standard metadata:
+    name => 'Mayriad\'s EH Master Script',
+    type => 'metadata',
+    namespace => 'memsplugin',
+    login_from => 'ehlogin',
+    author => 'Mayriad',
+    version => '1.0.0',
+    description => 'Accurately retrieves metadata from e-hentai.org using the identifiers appeneded to the filenames '
+      . 'of archives downloaded by Mayriad\'s EH Master Script.',
+    icon => 'data:image/png;base64,AAABAAEAEBAAAAEAIABoBAAAFgAAACgAAAAQAAAAIAAAAAEAIAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAD/'
+      . '//8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////AP///w'
+      . 'D///8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wARBmb/EQZm/xEGZv8RBmb/EQZm/////wD///8A////ABEG'
+      . 'Zv8RBmb/////AP///wARBmb/EQZm/////wD///8AEQZm/xEGZv8RBmb/EQZm/xEGZv////8A////AP///wARBmb/EQZm/////wD///8AEQZm/x'
+      . 'EGZv////8A////ABEGZv8RBmb/////AP///wD///8A////AP///wD///8AEQZm/xEGZv////8A////ABEGZv8RBmb/////AP///wARBmb/EQZm'
+      . '/////wD///8A////AP///wD///8A////ABEGZv8RBmb/////AP///wARBmb/EQZm/////wD///8AEQZm/xEGZv////8A////AP///wD///8A//'
+      . '//AP///wARBmb/EQZm/////wD///8AEQZm/xEGZv////8A////ABEGZv8RBmb/EQZm/xEGZv////8AEQZm/xEGZv////8AEQZm/xEGZv8RBmb/'
+      . 'EQZm/xEGZv8RBmb/////AP///wARBmb/EQZm/xEGZv8RBmb/////ABEGZv8RBmb/////ABEGZv8RBmb/EQZm/xEGZv8RBmb/EQZm/////wD///'
+      . '8AEQZm/xEGZv////8A////AP///wD///8A////AP///wARBmb/EQZm/////wD///8AEQZm/xEGZv////8A////ABEGZv8RBmb/////AP///wD/'
+      . '//8A////AP///wD///8AEQZm/xEGZv////8A////ABEGZv8RBmb/////AP///wARBmb/EQZm/////wD///8A////AP///wD///8A////ABEGZv'
+      . '8RBmb/////AP///wARBmb/EQZm/////wD///8AEQZm/xEGZv8RBmb/EQZm/xEGZv////8A////AP///wARBmb/EQZm/////wD///8AEQZm/xEG'
+      . 'Zv////8A////ABEGZv8RBmb/EQZm/xEGZv8RBmb/////AP///wD///8AEQZm/xEGZv////8A////ABEGZv8RBmb/////AP///wD///8A////AP'
+      . '///wD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////'
+      . 'AP///wD///8A////AP///wD///8A////AP///wD///8A//8AAP//AACDmQAAg5kAAJ+ZAACfmQAAn5kAAISBAACEgQAAn5kAAJ+ZAACfmQAAg5'
+      . 'kAAIOZAAD//wAA//8AAA==',
+    # Custom arguments:
+    oneshot_arg => 'Enter a valid EH gallery URL to copy metadata from this EH gallery to this LANraragi archvie',

```

Comment:
This duplicates functionality already in the stock E-H plugin (see review comment)